### PR TITLE
Warn when errorFields don't match request[[options]]

### LINKED ISSUE #793: Should we validate  nonsense payer error properties?
If we have a payment request that has option: `{requestPayerEmail: false}`, then what should we do if the merchant asks to correct something that was not requested, like 

```JS
response.retry({payer: {email: "Lol, I never asked for your email."} })
```

Right now, the spec just says to ignore that. Should we maybe throw or just continue to ignore. The above also applies to some cases of `.updateWith()` (e.g., flagging shipping address errors when no shipping address was requested). 

cc @aestes, @rsolomakhin, @romandev.

### PR DESCRIPTION
**Pull request by marcoscaceres**
closes #793

The following tasks have been completed:

 * [x] Confirmed there are no ReSpec errors/warnings.
 * [x] [Modified Web platform tests](https://github.com/web-platform-tests/wpt/pull/14246)

Implementation commitment:

 * [ ] Safari (link to issue)
 * [ ] [Chrome](https://bugs.chromium.org/p/chromium/issues/detail?id=909577)
 * [x] [Firefox](https://bugzilla.mozilla.org/show_bug.cgi?id=1494917)
 * [ ] Edge (public signal)

Optional, Impact on Payment Handler spec?

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/payment-request/pull/807.html" title="Last updated on Nov 28, 2018, 10:24 PM GMT (744302b)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/payment-request/807/614fa2a...744302b.html" title="Last updated on Nov 28, 2018, 10:24 PM GMT (744302b)">Diff</a>

### CODE REVIEW

**Comment by marcoscaceres** [comment]
@rsolomakhin, would mind having a first look?

**Review by rsolomakhin — APPROVED** [review]

**Comment by marcoscaceres** [comment]
Added manual test https://github.com/web-platform-tests/wpt/pull/14246

**Comment by marcoscaceres** [comment]
Anyone else willing to support these warnings (as we discussed in #793)? @aestes, @rsolomakhin?

File: index.html [hunk 1]
```diff
@@ -3111,26 +3111,63 @@ <h2>
...
+                    <var>errorFields</var>["<a data-link-for=
+                    "PaymentValidationErrors">payer</a>"]["<a data-link-for=
+                    "PayerErrors">phone</a>"] is present.
+                  </li>
+                  <li>
+                    <var>request</var>.<a>[[\options]]</a>["<a data-link-for=
+                    "PaymentOptions">requestShipping</a>"] is false, and
+                    <var>errorFields</var>["<a data-link-for=
+                    "PaymentValidationErrors">shippingAddress</a>"] has any
+                    members present.
```
Comment (domenic) [inline index.html]: Shouldn't this just be "is present"? As written now this will not warn if you pass `shippingAddress: {}`

File: index.html [hunk 2]
```diff
@@ -3111,26 +3111,63 @@ <h2>
...
-          type specified there. Otherwise, <a data-cite=
-          "WEBIDL#dfn-convert-ecmascript-to-idl-value">convert</a> to
-          <a data-cite="WEBIDL#idl-object">object</a>.
-          </li>
-          <li>If conversion results in a <a data-cite=
-          "WEBIDL#dfn-exception">exception</a> <var>error</var>:
+          <li>If <var>errorFields</var> was passed:
             <ol>
-              <li>Reject <var>retryPromise</var> with <var>error</var>.
+              <li>Optionally, show a warning in the developer console if any of
```
Comment (domenic) [inline index.html:3116]: FYI soon you and other specs will be able to link to a formal definition here, per https://github.com/whatwg/console/pull/151. No action for now though.

**Review by domenic — APPROVED** [review]
LGTM with question

File: index.html (hunk 2 above)
Comment (marcoscaceres) [reply index.html:3116]: Good to know! thanks!

File: index.html (hunk 1 above)
Comment (marcoscaceres) [reply index.html]: Yes, just being present is sufficient.

**Comment by marcoscaceres** [comment]
@rsolomakhin, filed [Chrome bug](https://bugs.chromium.org/p/chromium/issues/detail?id=909577). Hope that's ok, [based on discussion](https://github.com/w3c/payment-request/issues/793#issuecomment-426637553).

**Pull request merged**